Resources/ProxyProvider/DIRACCAProxyProvider.py

```diff
@@ -33,227 +66,249 @@ def __init__(self, parameters=None):
+    except (ValueError, KeyError) as e:
+      return S_ERROR('Unknown DN field in used DN: %s' % e)
+    nidOrder = [self.fields2nid[f] for f in self.dnList]
+    for index, nid in enumerate(userNIDs):
+      if nid not in nidOrder:
+        return S_ERROR('"%s" field not found in order.' % self.nid2field[nid])
+      if index > nidOrder.index(nid):
+        return S_ERROR('Bad DNs order')
+      for i in range(nidOrder.index(nid) - 1):
+        try:
+          if userNIDs.index(nidOrder[i]) > index:
+            return S_ERROR('Bad DNs order')
+        except (ValueError, KeyError):
+          continue
+      for i in range(nidOrder.index(nid) + 1, len(nidOrder)):
+        try:
+          if userNIDs.index(nidOrder[i]) < index:
+            return S_ERROR('Bad DNs order')
+        except (ValueError, KeyError):
+          continue
+
+    for nid in self.supplied:
+      if nid not in [self.fields2nid[f] for f in dnInfoDict]:
+        return S_ERROR('Current DN is invalid, "%s" field must be set.' % self.nid2field[nid])
+
+    for field, values in dnInfoDict.items():
+      nid = self.fields2nid[field]
+      err = 'Current DN is invalid, "%s" field' % field
+      if nid not in self.supplied + self.match + self.optional:
+        return S_ERROR('%s is not found for current CA.' % err)
+      if nid in self.match and not self.dnInfoDictCA[field] == values:
+        return S_ERROR('%s must be /%s=%s.' % (err, field,
+                                               ('/%s=' % field).joing(self.dnInfoDictCA[field])))
+      if nid in self.maxDict:
+        rangeMax = range(min(len(values), len(self.maxDict[nid])))
+        if any([True if len(values[i]) > self.maxDict[nid][i] else False for i in rangeMax]):
+          return S_ERROR('%s values must be less then %s.' % (err, ', '.join(self.maxDict[nid])))
+      if nid in self.minDict:
+        rangeMin = range(min(len(values), len(self.minDict[nid])))
+        if any([True if len(values[i]) < self.minDict[nid][i] else False for i in rangeMin]):
+          return S_ERROR('%s values must be more then %s.' % (err, ', '.join(self.minDict[nid])))
+
+      result = self.__fillX509Name(field, values)
       if not result['OK']:
         return result
-      if self.parameters['ProviderName'] not in result['Value']:
-        continue
-      dnDict = dict([field.split('=') for field in reqDN.lstrip('/').split('/')])
-      if not fullName:
-        fullName = dnDict.get('CN')
-      if not eMail:
-        eMail = dnDict.get('emailAddress')
-    self.log.info('Full name:', fullName)
-    self.log.info('Email:', eMail)
-    if not fullName:
-      return S_ERROR("Incomplete user information: no full name found")
-    if not eMail:
-      return S_ERROR("Incomplete user information: no email found")
 
     return S_OK({'Status': 'ready'})
 
-  def getProxy(self, userDict=None, sessionDict=None):
+  def getProxy(self, userDN):
```

Comment:
inconsistent signature with the other `ProxyProvider`

Reply:
now its the same